monitoring/bin/create_elasticsearch_datasource_tenant.sh

```diff
@@ -0,0 +1,135 @@
+        show_usage
+        exit 2
+      fi
+      ;;
+    -t|--tenant)
+      if [ -n "$2" ] && [ ${2:0:1} != "-" ]; then
+        tenant=$2
+        shift 2
+      else
+        log_error "A value for parameter [TENANT] has not been provided." >&2
+        show_usage
+        exit 2
+      fi
+      ;;
+    -u|--user)
+      if [ -n "$2" ] && [ ${2:0:1} != "-" ]; then
+        userID=$2
+        shift 2
+      else
+        # no initial user name provided, assign default name
+        log_error "A value for parameter [USER] has not been provided." >&2
+        show_usage
+        exit 2
+      fi
+      ;;
+    -p|--password)
+      if [ -n "$2" ] && [ ${2:0:1} != "-" ]; then
+        passwd=$2
+        shift 2
+      else
+        log_error "A value for parameter [PASSWORD] has not been provided." >&2
+        show_usage
+        exit 2
+      fi
+      ;;
+    -h|--help)
+      show_usage
+      exit
+      ;;
+    -*|--*=) # unsupported flags
+      log_error "Unsupported flag $1" >&2
+      show_usage
+      exit 1
+      ;;
+    *) # preserve positional arguments
+      POS_PARMS="$POS_PARMS $1"
+      shift
+      ;;
+  esac
+done
+
+# Convert namespace and tenant to all lower-case
+tenantNS=$(echo "$tenantNS"| tr '[:upper:]' '[:lower:]')
+tenant=$(echo "$tenant"| tr '[:upper:]' '[:lower:]')
+
+# Check to see if monitoring deployment could be found for tenant in provided namespace
+log_info "Checking the $tenantNS namespace for monitoring deployment for the $tenant tenant ..."
+if [[ $(kubectl get pods -n $tenantNS -l app.kubernetes.io/instance=v4m-grafana-$tenant -o custom-columns=:metadata.namespace --no-headers | uniq | wc -l) == 0 ]]; then
+    log_error "No monitoring components found in the $tenantNS namespace.  Monitoring needs to be deployed in this namespace in order to configure Elasticsearch with this script.";
+    exit 1
+else
+    log_verbose "Monitoring deployment was found for the $tenant tenant in the $tenantNS namespace.  Continuing."
+fi
+
+# Verify that the password is correct?
+
+
+# Create temporary directory for string replacement in the grafana-datasource-es.yaml file
+tenantDir=$TMP_DIR/$VIYA_TENANT
+mkdir -p $tenantDir
+cp -R monitoring/grafana-datasource-es.yaml $tenantDir/grafana-datasource-es.yaml
+
+# Replace placeholders
+log_debug "Replacing logging namespace for files in [$tenantDir]"
+if echo "$OSTYPE" | grep 'darwin' > /dev/null 2>&1; then
+  sed -i '' "s/__namespace__/$LOG_NS/g" $tenantDir/grafana-datasource-es.yaml
+  sed -i '' "s/__userID__/$userID/g" $tenantDir/grafana-datasource-es.yaml
+  sed -i '' "s/__passwd__/$passwd/g" $tenantDir/grafana-datasource-es.yaml
+else
+  sed -i "s/__namespace__/$LOG_NS/g" $tenantDir/grafana-datasource-es.yaml
+  sed -i "s/__userID__/$userID/g" $tenantDir/grafana-datasource-es.yaml
+  sed -i "s/__passwd__/$passwd/g" $tenantDir/grafana-datasource-es.yaml
+fi
+
+log_info "Provisioning Elasticsearch data source for Grafana"
```

Comment:
Message: Provisioning Elasticsearch data source for [$tenantNS/$tenant] instance of Grafana